src/c++/library/http_client.cc

```diff
@@ -1213,6 +1213,64 @@ InferenceServerHttpClient::ModelInferenceStatistics(
   return Get(request_uri, headers, query_params, infer_stat);
 }
 
+Error
+InferenceServerHttpClient::UpdateTraceSettings(
+    const std::string& model_name,
+    const std::map<std::string, std::vector<std::string>>& settings,
+    const Headers& headers, const Parameters& query_params)
+{
+  std::string request_uri(url_ + "/v2");
+  if (!model_name.empty()) {
+    request_uri += "/models/" + model_name;
+  }
+  request_uri += "/trace/setting";
+
+  triton::common::TritonJson::Value request_json(
+      triton::common::TritonJson::ValueType::OBJECT);
+  {
+    for (const auto& pr : settings) {
+      if (pr.second.empty()) {
+        request_json.Add(
+            pr.first.c_str(), std::move(triton::common::TritonJson::Value()));
+      } else {
+        if (pr.first == "trace_level") {
+          triton::common::TritonJson::Value level_json(
+              triton::common::TritonJson::ValueType::ARRAY);
+          for (const auto& v : pr.second) {
+            level_json.AppendString(v.c_str());
+          }
+          request_json.Add(pr.first.c_str(), std::move(level_json));
+        } else {
+          request_json.AddStringRef(pr.first.c_str(), pr.second[0].c_str());
+        }
+      }
+    }
+  }
+
+  triton::common::TritonJson::WriteBuffer buffer;
+  Error err = request_json.Write(&buffer);
+  if (!err.IsOk()) {
+    return err;
+  }
+
+  std::string response;
+  return Post(request_uri, buffer.Contents(), headers, query_params, &response);
+}
+
+Error
+InferenceServerHttpClient::GetTraceSettings(
+    std::string* settings, const std::string& model_name,
```

Comment:
Should return the parsed setting (something like `std::map<std::string, std::vector<std::string>>`) instead of the JSON string

Reply:
Although the other APIs are also simply returning the whole JSON string, perhaps we should keep it the same for consistency? @tanmayv25 what do you think

Reply:
I believe we must stick to returning settings as `std::string* settings`. 
Reason: The C++ client library should be a light-wrapper over the protocol. We give users flexibility to use json parser of their choice.